autode/conformers/conformers.py

```diff
@@ -1,111 +1,237 @@
 
-    mol_block_lines = Chem.MolToMolBlock(rdkit_mol_obj,
-                                         confId=conf_id).split('\n')
-    mol_file_atoms = []
+        return None
 
-    # Extract atoms from the mol block
-    for line in mol_block_lines:
-        split_line = line.split()
+    def prune_on_energy(self,
+                        e_tol:   Union[Energy, float] = Energy(1, units='kJ mol-1'),
+                        n_sigma: float = 5) -> None:
+        """
+        Prune the conformers based on an energy threshold, discarding those
+        that have energies that are similar to within e_tol. Also discards
+        conformers with very high energies (indicating a problem
+        with the calculation) if the are more than n_sigma standard deviations
+        away from the mean
 
-        if len(split_line) == 16:
-            atom_label = split_line[3]
-            x, y, z = split_line[0], split_line[1], split_line[2]
-            mol_file_atoms.append(Atom(atom_label, x=x, y=y, z=z))
+        Arguments:
+            e_tol (autode.values.Energy | float | None):
 
-    return mol_file_atoms
+            n_sigma (int): Number of standard deviations a conformer energy
+                   must be from the average for it not to be added
+        """
+        idxs_with_energy = [idx for idx, conf in enumerate(self)
+                            if conf.energy is not None]
+        n_prev_confs = len(self)
 
+        if len(idxs_with_energy) < 2:
+            logger.info(f'Only have {len(self)} conformers with an energy. No '
+                        f'need to prune')
+            return None
 
-def get_unique_confs(conformers, energy_threshold=Energy(1, units='kJ mol-1')):
-    """
-    For a list of conformers return those that are unique based on an energy
-    threshold in kJ mol^-1
+        energies = [self[idx].energy for idx in idxs_with_energy]
+        std_dev_e, avg_e = float(np.std(energies)), np.average(energies)
 
-    Arguments:
-        conformers (list(autode.conformer.Conformer)):
+        logger.info(f'Have {len(energies)} energies with μ={avg_e:.6f} Ha '
+                    f'σ={std_dev_e:.6f} Ha')
 
-        energy_threshold (autode.values.Energy): Energy threshold
+        if isinstance(e_tol, float):
+            logger.warning(f'Assuming energy tolerance {e_tol:.6f} has units '
+                           f'of Ha')
+            e_tol = Energy(e_tol, units='Ha')
 
-    Returns:
-        (list(autode.conformers.conformers.Conformer)): List of conformers
-    """
-    logger.info(f'Stripping conformers with energy ∆E < {energy_threshold} '
-                f'kJ mol-1 to others')
+        # Delete from the end of the list to preserve the order when deleting
+        for i, idx in enumerate(reversed(idxs_with_energy)):
+            conf = self[idx]
+            idxs_with_energy = [i for i in idxs_with_energy if i < len(self)]
+
+            if (np.abs(conf.energy - avg_e)
+                 / max(std_dev_e, 1E-8)) > n_sigma:
```

Comment:
i don't really like that you'll do this max comparison for every conformer

Reply:
I thought about this – probably not limiting but probably worth hoisting 👍